game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -202,64 +217,93 @@ def initialize_basic_iads(self) -> None:
         """Initialize the IADS Network in basic mode (SAM & EWR only)"""
         for go in self.ground_objects.values():
             if isinstance(go, IadsGroundObject):
-                try:
-                    self.node_for_tgo(go)
-                except IadsNetworkException:
+                node = self.node_for_tgo(go)
+                if node is None:
                     # TGO does not participate to the IADS -> Skip
                     pass
 
     def initialize_network_from_config(self) -> None:
         """Initialize the IADS Network from a configuration"""
         for element_name, connections in self.iads_config.items():
+            warning_msg = (
+                f"IADS: No ground object found for {element_name}."
+                f" This can be normal behaviour."
+            )
             try:
                 node = self.node_for_tgo(self.ground_objects[element_name])
-            except (KeyError, IadsNetworkException):
+            except KeyError:
+                node = None
+                warning_msg = (
+                    f"IADS: No ground object found for connection {element_name}"
+                )
+
+            if node is None:
                 # Log a warning as this can be normal. Possible case is for example
                 # when the campaign request a Long Range SAM but the faction has none
                 # available. Therefore the TGO will not get populated at all
-                logging.warning(
-                    f"IADS: No ground object found for {element_name}. This can be normal behaviour."
-                )
+
+                logging.warning(warning_msg)
                 continue
 
             # Find all connected ground_objects
             for node_name in connections:
                 try:
-                    node.add_connection_for_tgo(self.ground_objects[node_name])
-                except (KeyError):
+                    tgo = self.ground_objects[node_name]
+                except KeyError:
                     logging.error(
                         f"IADS: No ground object found for connection {node_name}"
                     )
                     continue
+                if tgo is None:
+                    emsg = f"IADS: ground_objects contains None values:\n{tgo}"
+                    raise IadsNetworkException(emsg)
```

Comment:
This is another situation of which I believe we should be notified, because this shouldn't be possible to occur...

Reply:
Well, mypy complains this code is unreachable, I seriously hope it's right because the only way to fix it is to delete it -_-